internal/provider/models/deployment.go

```diff
@@ -577,3 +587,32 @@ func ScalingSpecTypesObject(
 	}
 	return types.ObjectValueFrom(ctx, schemas.ScalingSpecAttributeTypes(), obj)
 }
+
+type RemoteExecution struct {
+	Enabled                types.Bool   `tfsdk:"enabled"`
+	AllowedIpAddressRanges types.Set    `tfsdk:"allowed_ip_address_ranges"`
+	RemoteApiUrl           types.String `tfsdk:"remote_api_url"`
+	TaskLogBucket          types.String `tfsdk:"task_log_bucket"`
+	TaskLogUrlPattern      types.String `tfsdk:"task_log_url_pattern"`
+}
+
+func RemoteExecutionTypesObject(
+	ctx context.Context,
+	remoteExecution *platform.DeploymentRemoteExecution,
+) (types.Object, diag.Diagnostics) {
+	if remoteExecution == nil {
+		return types.ObjectNull(schemas.RemoteExecutionAttributeTypes()), nil
+	}
+	allowedIpAddressRanges, diags := utils.StringSet(&remoteExecution.AllowedIpAddressRanges)
+	if diags.HasError() {
+		return types.ObjectNull(schemas.RemoteExecutionAttributeTypes()), diags
+	}
+	obj := RemoteExecution{
+		Enabled:                types.BoolValue(remoteExecution.Enabled),
+		AllowedIpAddressRanges: allowedIpAddressRanges,
+		RemoteApiUrl:           types.StringValue(remoteExecution.RemoteApiUrl),
```

Comment:
just trying to learn the ropes: Is this request struct being generated, or API response being converted to an object for terraform SDK to handle?
For requests, users don't need to pass remote API url

Reply:
this is the latter, the API response is being converted into an object for terraform to handle and read